File: intro-prog/lista 3_extra/ex9.py
"""
9. Escreva um programa que, dado um número inteiro positivo, o imprime em notação binária. O
processo de obtenção é similar à extrair os dígitos de um número: divisões sucessivas por 2, até
que o quociente seja 0 (zero). A cada divisão, o resto indica um dígito binário. Entretanto, assim
como no algoritmo de obter os dígitos, os dígitos binários estarão em ordem inversa. Logo, será
preciso remontar o número binário (na ordem correta) em uma variável inteira. Observe o exemplo:
"""

n = int(input("Digite um numero "))
binario = invertido = digito = 0

while n != 0:
    digito = n % 2
    invertido = invertido * 10 + digito
    n = n // 2

while invertido != 0:
    digito = invertido % 10
    binario = binario * 10 + digito
    invertido //= 10

print(f"{binario}")

File: Algoritmos/Lista 1/ex9.c
/*9. Escreva uma função que recebe três números e os imprime em ordem crescente.*/

#include <stdio.h>

void print_sorted(int a, int b, int c){
    if (a <= b && b <= c) {
    printf("%d %d %d\n", a, b, c);
    }
    else if (a <= c && c <= b) {
        printf("%d %d %d\n", a, c, b);
    }
    else if (b <= a && a <= c) {
        printf("%d %d %d\n", b, a, c);
    }
    else if (b <= c && c <= a) {
        printf("%d %d %d\n", b, c, a);
    }
    else if (c <= a && a <= b) {
        printf("%d %d %d\n", c, a, b);
    }
    else if (c <= b && b <= a) {
        printf("%d %d %d\n", c, b, a);
    }


}


int main(){
    int a, b ,c;

    printf("Digite o primeiro numero: ");
    scanf("%d", &a);
    printf("Digite o segundo numero: ");
    scanf("%d", &b);
    printf("Digite o terceiro numero: ");
    scanf("%d", &c);

    print_sorted(a, b, c);
    return 0;
}